FORMS.md

```diff
@@ -0,0 +1,243 @@
+  <aside class="ods-field--error" id="first-name-error"><span class="ods-u-visually-hidden">Error:</span> Numbers are disallowed.</aside>
+</fieldset>
+```
+
+## FieldGroup
+
+### Non-global attrs
+
+Fieldset-specific HTML attributes:
+
+- `disabled`: If this Boolean attribute is set, all Inputs that are descendants of the `<fieldset>` are disabled.
+  - Form elements inside the `<legend>` element won't be disabled, but this doesn't seem like a valid use case for Odyssey.
+  - In the case of FieldGroup, `disabled` could be passed to the child Fields (or their Inputs). Disabling whole FieldGroups on the parent `fieldset` is also viable.
+- `name`: The name associated with the group.
+- `form`: This attribute takes the value of the id attribute of a `<form>` element you want the `<fieldset>` to be part of, even if it is not inside the form. Please note that usage of this is confusing — if you want the `<input>` elements inside the `<fieldset>` to be associated with the form, you need to use the form attribute directly on those elements.
+  - Because of the unintuitive behavior of this attribute, I recommend that this be relegated to use on Field or we provide a way to cascade this attribute to all child Inputs.
+
+### Content areas
+
+- Title: contains legend for FieldGroup
+- Descr: contains a description for a FieldGroup
+- Slot: contains Fields
+
+### Pseudo-Structure
+
+```html
+<FieldGroup>
+  <FieldGroup.Title>A Group of Fields</FieldGroup.Title>
+  <FieldGroup.Desc>This is a description of the FieldGroup.</FieldGroup.Desc>
+  [...]
+</FieldGroup>
+```
+
+### HTML
+
+```html
+<fieldset class="ods-field-group">
+  <legend class="ods-field-group--title">A Group of Fields</legend>
+  <p class="ods-field-group--desc">This is a description of the FieldGroup.</p>
+  [...]
+</fieldset>
+```
+
+## Form
+
+A container component that controls form behavior and provides layout slots.
+
+### Non-global attrs
+
+Form-specific HTML attributes we may need props for.
+
+#### Submission-related
+
+These values can be overridden by a `formaction` attribute on a `<button>`, `<input type="submit">`, or `<input type="image">` element.
+
+- `action`: The URL that processes the form submission.
+- `enctype`: If the value of the method attribute is post, enctype is the MIME type of the form submission. Possible values: `application/x-www-form-urlencoded` (default), `multipart/form-data` (use when form contains `<input type='file'>` elements), `text/plain` (debugging).
+- `method`: The HTTP method to submit the form with. Possible (case insensitive) values: `get`, `post`, `dialog`.
+- `novalidate`: This Boolean attribute indicates that the form shouldn't be validated when submitted.
+- `target`: Indicates where to display the response after submitting the form. The following values have special meanings: `_self` (default), `_blank`, `_parent`, `_top`.
```

Comment:
These are standard attributes, yeah? I don't think we should explicitly define props (attributes) that are part of the standard, but rather allow these types of attributes to be spread onto the component. I agree that we should allow developers to use these if they need to.

Let's definitely talk about what the React component API could look like and map that out! I'm curious how we can use the [React Context API ](https://reactjs.org/docs/context.html) here as well. Let's talk about this tomorrow, I have some ideas.

Reply:
Agreed! I don't mean to suggest that all available attributes should be converted into props. Rather, I want to surface all of the available behaviors we may want to make available (or enforce) for the component.

e.g. Maybe we want to disallow certain `target` values or enforce an `enctype` when Input types are detected.

Reply:
Good feedback - I'm struggling a bit to understand in this doc where the HTML standards end and the component API design begins

Reply:
This doc is more opinionated about rendered output and component structure than React implementation. The goal of surfacing the associated HTML attributes and structure is to ensure that they're accounted for in a consistent manner in the React implementation, which has been historically lacking in our products.

That is, attributes like `id` associations or `aria-invalid` need to be present, but authors should not be required to address them manually.

This should also provide appropriate guidance for those who are not using our React implementation or when building non-React components in the future.